app/controllers/teacher_interface/further_information_request_items_controller.rb

```diff
@@ -4,8 +4,44 @@ class FurtherInformationRequestItemsController < BaseController
 
     def edit
       @further_information_request_item = further_information_request_item
-      @further_information_request =
-        further_information_request_item.further_information_request
+      @further_information_request = further_information_request
+
+      if further_information_request_item.text?
+        @further_information_request_item_form =
+          FurtherInformationRequestItemTextForm.new(
+            response: further_information_request_item.response,
+          )
+      end
+    end
+
+    def update
+      if further_information_request_item.text?
+        @further_information_request_item_form =
+          FurtherInformationRequestItemTextForm.new(
+            further_information_request_form_params.merge(
+              further_information_request_item:,
+            ),
+          )
+      end
+
+      if params[:next] == "save_and_continue"
+        if @further_information_request_item_form.save
```

Comment:
Will this be `nil` if further_information_request_item isn't text? I think this action is getting a bit out of hand in any case. Could we move some of this logic into something that handles any further_information_request_item type and move all the branching out into that?